scipy/optimize/tests/test_optimize.py

```diff
@@ -2180,6 +2181,33 @@ def func(z, *params):
                           match=r'Either final optimization did not succeed'):
             optimize.brute(func, self.rranges, args=self.params, disp=True)
 
+        message = "Polishing is incompatible with integrality constraints"
+        with pytest.warns(OptimizeWarning, match=message):
+            optimize.brute(func, self.rranges, args=self.params,
+                           integrality=True)
+
```

Comment:
No test for no-repeat integers between `(low, high)`.

Reply:
The original logic ensured that there would not be repeated integers by reducing `Ns` if needed.
Now, the logic uses `np.unique` explicitly.